core/src/androidTest/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/YouTubePlayerViewClickTest.kt

```diff
@@ -0,0 +1,81 @@
+package com.pierfrancescosoffritti.androidyoutubeplayer.core
+
+import android.view.MotionEvent
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.filters.LargeTest
+import androidx.test.rule.ActivityTestRule
+import com.pierfrancescosoffritti.androidyoutubeplayer.test.R
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.helpers.MotionEventPrimitive
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.testActivity.TestActivity
+import org.junit.Assert
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+
+@LargeTest
+class YouTubePlayerViewClickTest {
+
+    @Rule @JvmField
+    var activityRule = ActivityTestRule(TestActivity::class.java)
+
+    @Test
+    fun testSingleClickListener() {
+        onView(withId(R.id.youtube_player_view)).perform(click())
```

Comment:
So...the click test and longClick test use mockito now. But the touch listeners tests (swipeRight and swipeDown) need implementation detail (since they're not a test to just see if a method is called or not), i'm currently using the manual method for these tests. If there's a better way, please let me know so I can improve it.